src/distPlus/distribution/XYShape.re

```diff
@@ -137,6 +142,61 @@ module XtoY = {
       };
     n;
   };
+
+  /* The extrapolators below can be used e.g. with PointwiseCombination.combine */
+  let linearBetweenPointsExtrapolateFlat = (t: T.t, leftIndex: int, x: float) => {
+    if (leftIndex < 0) {
+      t.ys[0];
+    } else if (leftIndex >= T.length(t) - 1) {
+      T.lastY(t);
+    } else {
+      let x1 = t.xs[leftIndex];
+      let x2 = t.xs[leftIndex + 1];
+      let y1 = t.ys[leftIndex];
+      let y2 = t.ys[leftIndex + 1];
+      let fraction = (x -. x1) /. (x2 -. x1);
+      y1 *. (1. -. fraction) +. y2 *. fraction;
+    }
+  };
+
+  let linearBetweenPointsExtrapolateZero = (t: T.t, leftIndex: int, x: float) => {
+    if (leftIndex < 0) {
+      0.0
+    } else if (leftIndex >= T.length(t) - 1) {
+      0.0
+    } else {
+      let x1 = t.xs[leftIndex];
+      let x2 = t.xs[leftIndex + 1];
+      let y1 = t.ys[leftIndex];
+      let y2 = t.ys[leftIndex + 1];
+      let fraction = (x -. x1) /. (x2 -. x1);
+      y1 *. (1. -. fraction) +. y2 *. fraction;
```

Comment:
This looks very similar to the code above. Given it's this long I would imagine it should be dried up.